NX and mikro-orm CLI - cannot find module (tsconfig: compilerOptions > path

**Describe the bug**
A bug that the mikro-orm cli has when being used in the nx monorepo environment (and most likely other environments using the compilerOption > paths option in the tsconfig)

The problem could be that exporting a `type` is alright by TypeScript, but exporting a `const` is not. However, when running (and building the code) there is no problem with this.

I created a reproduction repository with a similar explanation. Also with steps how to run it and reproduce the bug.

https://github.com/RiskChallenger/nx-mikro-orm-cli

**Stack trace**
```
- configuration not found (Cannot find module '@test-mikro-orm/shared'
Require stack:
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/house.entity.ts
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/project.entity.ts
- /home/langstra/Documents/test-mikro-orm/apps/api/src/app/mikro-orm.config.ts
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/utils/ConfigurationLoader.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/utils/index.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/typings.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/core/index.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/cli/CLIConfigurator.js
- /home/langstra/Documents/test-mikro-orm/node_modules/@mikro-orm/cli/cli.js)
Done in 0.60s.
```

**To Reproduce**
Steps to reproduce the behavior:
```bash
$ git clone git@github.com:RiskChallenger/nx-mikro-orm-cli.git
$ cd nx-mikro-orm-cli
$ yarn
$ yarn mikro-orm debug
```


**Versions**
```bash
Current MikroORM CLI configuration
 - dependencies:
   - mikro-orm 4.5.4
   - node 14.4.0
   - typescript 4.1.5
 - package.json found
 - ts-node enabled
 - searched config paths:
   - /home/langstra/Documents/test-mikro-orm/apps/api/src/app/mikro-orm.config.ts (found)
   - /home/langstra/Documents/test-mikro-orm/mikro-orm.config.ts (not found)
   - /home/langstra/Documents/test-mikro-orm/mikro-orm.config.js (not found)

```

--- a/packages/core/src/utils/ConfigurationLoader.ts
+++ b/packages/core/src/utils/ConfigurationLoader.ts
@@ -1,6 +1,6 @@
 import dotenv from 'dotenv';
 import { pathExists, readFile } from 'fs-extra';
-import { join, isAbsolute } from 'path';
+import { dirname, join, isAbsolute } from 'path';
 import stripJsonComments from 'strip-json-comments';
 import { IDatabaseDriver } from '../drivers';
 import { Configuration, Options } from './Configuration';
@@ -90,8 +90,18 @@ export class ConfigurationLoader {
   }
 
   static async getTsConfig(tsConfigPath: string): Promise<Dictionary> {
-    const json = await readFile(tsConfigPath);
-    return JSON.parse(stripJsonComments(json.toString()));
+    const tsConfigFile = await readFile(tsConfigPath);
+    const tsConfig = JSON.parse(stripJsonComments(tsConfigFile.toString()));
+
+    if (!Object.keys(tsConfig).includes('extends')) {
+      return tsConfig;
+    }
+
+    const tsConfigFolderPath = dirname(tsConfigPath);
+    const tsConfigExtendsPath = isAbsolute(tsConfig.extends) ? tsConfig.extends : join(tsConfigFolderPath, tsConfig.extends);
+    const baseTsConfig = await ConfigurationLoader.getTsConfig(tsConfigExtendsPath);
+
+    return Utils.merge(baseTsConfig, tsConfig);
   }
 
   static loadEnvironmentVars<D extends IDatabaseDriver>(options?: Options<D> | Configuration<D>): Partial<Options<D>> {

--- a/tests/features/cli/CLIHelper.test.ts
+++ b/tests/features/cli/CLIHelper.test.ts
@@ -6,7 +6,14 @@ jest.mock(process.cwd() + '/mikro-orm-async.config.js', () => (Promise.resolve({
 jest.mock(process.cwd() + '/mikro-orm-async-catch.config.js', () => (Promise.reject('FooError')), { virtual: true });
 const pkg = { 'mikro-orm': {} } as any;
 jest.mock(process.cwd() + '/package.json', () => pkg, { virtual: true });
-const tsc = { compilerOptions: {} } as any;
+
+const tscBase = { compilerOptions: { baseUrl: '.', paths: { '@some-path/some': './libs/paths' } } } as any;
+jest.mock(process.cwd() + '/tsconfig.base.json', () => tscBase, { virtual: true });
+
+const tscExtended = { extends: './tsconfig.base.json', compilerOptions: { module: 'commonjs' } } as any;
+jest.mock(process.cwd() + '/tsconfig.json', () => tscExtended, { virtual: true });
+
+const tsc = { compilerOptions: { } } as any;
 jest.mock(process.cwd() + '/tsconfig.json', () => tsc, { virtual: true });
 
 import c from 'ansi-colors';
@@ -71,25 +78,22 @@ describe('CLIHelper', () => {
     requireFromMock.mockRestore();
   });
 
-  test('configures yargs instance [ts-node and ts-paths]', async () => {
-    const readFileMock = jest.spyOn(require('fs-extra'), 'readFile');
-    readFileMock.mockImplementation(async () => JSON.stringify(tsc));
+  test('configures yargs instance [ts-node and ts-paths and tsconfig.extends]', async () => {
     const pathExistsMock = jest.spyOn(require('fs-extra'), 'pathExists');
     pathExistsMock.mockResolvedValue(true);
     pkg['mikro-orm'].useTsNode = true;
-    tsc.compilerOptions.paths = { alternativePath: ['alternativePath'] };
     const requireFromMock = jest.spyOn(Utils, 'requireFrom');
     requireFromMock
       .mockImplementationOnce(() => ({ register: jest.fn() }))
       .mockImplementationOnce(() => ({ register: jest.fn() }));
+    const tsconfigExtendsMock = jest.spyOn(ConfigurationLoader, 'getTsConfig');
     const cli = await CLIConfigurator.configure() as any;
     expect(cli.$0).toBe('mikro-orm');
     expect(requireFromMock).toHaveBeenCalledWith('ts-node', process.cwd() + '/tsconfig.json');
     expect(requireFromMock).toHaveBeenCalledWith('tsconfig-paths', process.cwd() + '/tsconfig.json');
+    expect(tsconfigExtendsMock).toHaveBeenCalledTimes(2);
     pathExistsMock.mockRestore();
     pkg['mikro-orm'].useTsNode = false;
-    delete tsc.compilerOptions.paths;
-    readFileMock.mockRestore();
     requireFromMock.mockRestore();
   });
 

